testsuite/testclient/conftest.py

```diff
@@ -0,0 +1,900 @@
+"""
+Module for running the testclient .yaml test cases.
+"""
+
+import doctest
+import socket
+import pytest
+import re
+import traceback
+import threading
+from collections import namedtuple
+try:
+    from collections import OrderedDict
+except ImportError:
+    from ordereddict import OrderedDict
+import yaml
+import yamlordereddictloader
+import httpretty
+from httpretty.core import HTTPrettyRequestEmpty, fakesock
+from lxml import etree, doctestcompare
+import six
+import pywbem
+from pywbem._utils import _ensure_unicode
+from pywbem._nocasedict import NocaseDict
+
+
+class ExcThread(threading.Thread):
+    """
+    Thread that catches an exception and passes it to the creator
+    in the Thread.join() method.
+    """
+
+    def run(self):
+        self.exc = None
+        try:
+            threading.Thread.run(self)
+        except Exception:
+            import sys
+            self.exc = sys.exc_info()
+
+    def join(self, *args, **kwargs):
+        threading.Thread.join(self, *args, **kwargs)
+        if self.exc:
+            six.reraise(*self.exc)
+
+
+def patched_makefile(self, mode='r', bufsize=-1):
+    """Returns this fake socket's own StringIO buffer.
+
+    If there is an entry associated with the socket, the file
+    descriptor gets filled in with the entry data before being
+    returned.
+    """
+    self._mode = mode
+    self._bufsize = bufsize
+
+    if self._entry:
+        # The patched version uses ExcThread instead of Thread
+        t = ExcThread(
+            target=self._entry.fill_filekind, args=(self.fd,)
+        )
+        t.start()
+        if self.timeout == socket._GLOBAL_DEFAULT_TIMEOUT:
+            timeout = None
+        else:
+            timeout = self.timeout
+        t.join(timeout)
+        if t.isAlive():
+            raise socket.timeout
+
+    return self.fd
+
+
+# Monkey-patching httpretty to pass exception raised in callbacks
+# back to the caller.
+fakesock.socket.makefile = patched_makefile
+
+
+def pytest_collect_file(parent, path):
```

Comment:
That one took a while to sort out until I finally found it in the pytest documentation. Think it needs more explanation than "pytest hook"

Reply:
Since I assume you derived  this from 
https://docs.pytest.org/en/latest/example/nonpython.html#a-basic-example-for-specifying-tests-in-yaml-files

Lets point back to it.